packages/app/src/hooks/useMetadata.tsx

```diff
@@ -115,3 +122,21 @@ export function useGetKeyValues({
     placeholderData: keepPreviousData,
   });
 }
+
+function deduplicate2dArray<T extends object>(array2d: T[][]): T[] {
+  const array: T[] = [];
+  const set = new Set<string>();
+  for (const _array of array2d) {
+    for (const elem of _array) {
+      const key = objectHash.sha1(elem);
+      if (set.has(key)) continue;
+      set.add(key);
+      array.push(elem);
+    }
+  }
+  return array;
+}
+
+// export functions for testing only
+export const testExports =
+  process.env.NODE_ENV === 'test' ? { deduplicate2dArray } : undefined;
```

Comment:
we probably don't set the node env for unit test (also you want to use `IS_CI` flag instead). I think we can just export this function